oslabs-beta/Kubiquity
# Update electron

**WHAT DOES THIS PR DO?**
1. Converts our MERN app to a full Electron app. 
2. Various style and name changes. 

**HOW TO TEST THIS PR:**
1. Run: `gh pr checkout 29 && npm install && npm run build && npm start`
2. Verify that the app works as expected, then kill the app. 
3. Follow the instructions in `App.jsx` to convert the app from the testing environment to the K8s environment. 
4. Open Docker Desktop. 
5. Run `minikube start`
6. Run `helm upgrade -i prometheus prometheus-community/prometheus --namespace prometheus`
7. Run `npm run build && npm start`
8. Verify that the app works as expected. 

**ADDITIONAL NOTES:**
1. What has changed?? We are no longer running an Express server on the backend. 
2. Why is the server still in the directory?? This is for backend—they can decide if they want it or not, and remove from the project as needed. 
3. What still needs to be done with Electron?? We need to bundle up the work and export it, but this isn't too difficult and will come in a later PR. 
4. Why do I have to refresh the app to get the table to load?? We're still using MongoDB, which means we still have to connect to our DB, which means there will still be some lag between the app launching and mongoDB connecting. However, because we're going to switch over to some local storage, this will not be an issue in the near future.
